src/data/finsupp/basic.lean

```diff
@@ -774,6 +775,17 @@ end
 @[simp] lemma graph_eq_empty {f : α →₀ M} : f.graph = ∅ ↔ f = 0 :=
 (graph_injective α M).eq_iff' graph_zero
 
+/-- Produce an association list for the finsupp over its support using choice. -/
+@[simps] def to_alist (f : α →₀ M) : alist (λ x : α, M) :=
+⟨f.graph.to_list.map prod.to_sigma, begin
+  rw [list.nodupkeys, list.keys, list.map_map, prod.fst_comp_to_sigma, list.nodup_map_iff_inj_on],
+  { rintros ⟨b, m⟩ hb ⟨c, n⟩ hc (rfl : b = c),
+    rw [mem_to_list, finsupp.mem_graph_iff] at hb hc,
+    dsimp at hb hc,
+    rw [←hc.1, hb.1] },
+  { apply nodup_to_list }
+end⟩
```

Comment:
What happened to the `graph_nodupkeys` lemma you had in #15443? I think it might make sense to just leave this all in one PR, about conversions between finsupp and alist.

Reply:
My idea was to separate `finsupp.to_alist` and then prove some results separately, but I couldn't find anything that didn't depend on the new definitions in #15443. I'll close this PR and just leave the original one.